### ISSUE TITLE
Get Accessibility Caption that is a part of GraphImage metadata

### ISSUE THREAD

**Issue by chrisndeca** [arc298/instagram-scraper#608]
There is an Accessibility Caption in the IG Metadata, is there anyway to capture that?

Thanks

**Comment by do-me** [comment]
I tried to implement it but for how I understood you cannot find the accessibility caption in the scraped json (variable "item"). An after-scrape workaround could be to iterate over all post ids afterwards. In Python you could set up a pandas dataframe and append it for every post id with: 
```
json.loads(requests.get("https://www.instagram.com/p/[yourpostid]/?__a=1").text)["graphql"]["shortcode_media"]`
```
For anyone excited about fixing it: give pdb a try. It's Python's debugging console. Just put these lines in Instagram-Scraper's "app.py" wherever you'd like to investigate a variable: 

```
import pdb
from pdb import set_trace as bp
bp()
```

**Comment by do-me** [comment]
For anyone still interested in mining accessibility captions and post metadata such as location IDs I wrote Python-Selenium-based [Simple Instagram Scraper](https://github.com/do-me/Simple-Instagram-Scraper). You should be able to scrape up to 10k posts with all information (until now just metadata implemented, no media download) at a rate of 0.1-0.2 posts/second or 5-10 seconds/post.

If you are only interested in getting post metadata (excluding location and accessibility caption) **fast** check out Python-Torpy-based [Fast Instagram Scraper](https://github.com/do-me/fast-instagram-scraper) accessing the same data as Instagram Scraper only way faster at a rate of 5 posts/second or 0.2 seconds/post.

**Comment by chrisndeca** [comment]
Thank you! Very nice!

**Comment by monajalal** [comment]
@do-me what if we want to scrape more than 10k posts like 1M posts? Also, is there anyway to download all the followings at the same time for a given hashtag?
1. image
2. caption
3. accessibility_captions
4. comments

for me 1, 2, and 3 are pretty much more important currently. Thanks for your guidance.

**Comment by do-me** [comment]
@monajalal Unfortunately I don't see any way of mining accessibility_captions for this amount of posts. For a quick overview refer to this table https://github.com/do-me/fast-instagram-scraper#scraper-comparison or to my blog https://geo.rocks/post/writing-your-own-instagram-scraper/ where the tradeoff between speed and level of detail is explained (also applies to Instagram-Scraper). 

The reason why you cannot mine accessibility_captions at scale is the necessary login for mining. Instagram Scraper currently cannot mine these captions as they are not included in the delivered JSON files. [Simple Instagram Scraper]https://github.com/do-me/Simple-Instagram-Scraper) can mine these captions as it visualizes every single post. This works only when logged in or when using residential proxies.
In case, you could try to set the break times for [Simple Instagram Scraper]https://github.com/do-me/Simple-Instagram-Scraper) so high that you don't run into any Instagram limitation. Probably this would take way too much time still.
Until now I didn't implement downloading any media yet but this would probably be a simple one-liner. 

However you could also combine both, Instagram Scraper for donwloading media and [Simple Instagram Scraper]https://github.com/do-me/Simple-Instagram-Scraper) for accessibility_captions and location IDs.

If you find any other good solution Let us know!
